cpp/include/raft/core/logger-inl.hpp

```diff
@@ -152,4 +152,16 @@ RAFT_INLINE_CONDITIONAL void logger::flush() { pimpl->spdlogger->flush(); }
 
 RAFT_INLINE_CONDITIONAL logger::~logger() {}
 
+struct log_level_setter {
+  explicit log_level_setter(int level)
+  {
+    prev_level_ = logger::get(RAFT_NAME).get_level();
+    logger::get(RAFT_NAME).set_level(level);
+  }
+  ~log_level_setter() { logger::get(RAFT_NAME).set_level(prev_level_); }
+
+ private:
+  int prev_level_;
+};
```

Comment:
Sorry, I should have given this a bit more thought in my previous comment. People who use the raft::compiled library will not be able to access `log_level_setter` (because they only include logger-ext.hpp).

The "perfect" solution would be to split up log_level_setter into a declaration (logger-ext.cuh) and a definition (logger-inl.cuh), like was done with logger.

The practical solution is to put log_level_setter in logger-ext.cuh. Then consumers of both `raft::compiled` and `raft::header_only` (?) will see log_level_setter. The compile time overhead of this struct is close to zero, so that should be fine. 

Background: the reason this header was split is that the logger-inl.cuh includes spdlog, which is a beast to compile.

Reply:
No worries! Thanks for catching it before this merged. Updating accordingly